### DISCUSSION TITLE
How to bind ColumnClause to FromClause?

### DISCUSSION THREAD

**Discussion by apabolleta-dasnano** [sqlalchemy/sqlalchemy#11601] [Usage Questions]
Is it possible to bind generic ColumnClause objects to specific FromClause object, so can dynamically select columns in a selectable object without explicit reference to it?

An example of use:

```
import sqlalchemy as sa

def get_select(selectable, *columns):
    # bind columns to selectable here...

columns = (sa.Column('col1'), sa.Column('col2').label('col_2'), sa.Column('col3'))

my_select = get_select(my_table, *columns)
```

**Comment by CaselIT** [comment]
Hi,

You would use `column` lowercase for that. The upper case one may work too, but I'm not sure if you can re-use them.
Then you can do `select(*the_columns).select_from(the_table)`

**Reply by apabolleta-dasnano** [reply]
Thanks for the answer.

Both approaches work well and produce valid SQL. However, column names generated are not prefixed with table name `'<tablename>.<columname>'`, as when selected from `table.c`.

**Reply by CaselIT** [reply]
Well you haven't assigned any table to those columns. 
You may try to use https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.Select.set_label_style to change that, but I'm not sure if it helps in this case
